### PR TITLE
Spree upgrade step 6

### PR DESCRIPTION
**Pull request by Em-AK**
This branch reflects the ongoing work for the spree upgrade.

Step branches are created as described here : https://community.openfoodnetwork.org/t/progress-report-for-spree-upgrade/615

When we think a step is ready we merge it in here. Every few (~5) steps this branch should be deployed on a staging server for manual testing before merging into upstream master.

### CODE REVIEW

**Comment by RohanM** [comment]
@Em-AK @enricostano Nice work! I think we're planning on staging and testing this branch soon. Would you be able to rebase to fix the `Gemfile.lock` conflict?

**Comment by enricostano** [comment]
@RohanM rebased, thanks!

File: app/helpers/spree/admin/base_helper_decorator.rb [hunk 1]
```diff
@@ -31,8 +31,7 @@ def link_to_remove_fields_without_url(name, f, options = {})
         html_options = {class: "remove_fields #{options[:class]}", data: {action: 'remove'}, title: t(:remove)}
         html_options.merge!(options[:html]) if options.key? :html
 
-        link_to_with_icon('icon-trash', name, '#', html_options).gsub('href="#" ', '').html_safe +
-          f.hidden_field(:_destroy)
+        link_to_with_icon('icon-trash', name, '#', html_options).gsub('href="#" ', '') + f.hidden_field(:_destroy)
```
Comment (sauloperez) [inline app/helpers/spree/admin/base_helper_decorator.rb]: any reason why the `html_safe` got removed?

Comment (enricostano) [inline app/helpers/spree/admin/base_helper_decorator.rb]: it's bad practice, you should use Rails tag helpers, as we're already doing

**Comment by enricostano** [comment]
All green again 🍏 let's test that we solved all the issues! 💪 

It is deployed at https://staging1.openfood.com.au

cc/ @sstead @sigmundpetersen @OliverUK

**Comment by enricostano** [comment]
@mkllnk did you rebased this branch against the recent `master`? I see there is a [new red spec](https://travis-ci.org/openfoodfoundation/openfoodnetwork/jobs/270202283#L2827) in Travis that I didn't see before. Any idea?

**Comment by mkllnk** [comment]
@enricostano Yes, I did.

**Comment by enricostano** [comment]
OMG 😱

**Comment by NickWeir63** [comment]
Thanks so much @enricostano @sauloperez and everyone who has made this happen.  Massive piece of work.  Huge appreciation :)

**Pull request merged**